File: Assignment_4/cpp/flyweight/driver.cpp
#include <iostream>
#include "Document.h"

int main(int argc, char*argv[])
{
  constexpr char document_name[] = "HelloWorldCS5800";

  std::cout << "Create text document: " << document_name << std::endl;

  Document document(document_name);
  document.AddText("Test ", FontType::Arial, 12, ColorType::Black());
  document.AddText("Done! ", FontType::Verdana, 16, ColorType::Black());

  document << " to see if this works!";
  document << std::make_tuple<>(FontType::Calibri, 18, ColorType::Black());
  document << " This text is a different size and font!";
  document << std::make_tuple<>(FontType::Verdana, 16, ColorType::Black());
  document << " This text should use an already used TextType!";

  std::cout << "save document: " << document.GetFilename() << "..." << std::endl;
  document.Save();
  std::cout << "document " << document.GetFilename() << " saved!" << std::endl;

  return 0;
}

File: Assignment_4/java/flyweight/src/Driver.java
import java.io.IOException;

public class Driver {
    public static void main(String[] args) throws IOException {
        Document document = new Document("HelloWorldCS5800");

        document.AddText("This is the first Test", FontType.ARIAL, 12, ColorType.Black());
        document.AddText(" Test IS 2nd", FontType.ARIAL, 12, ColorType.Black());
        document.AddText(" What 3rd?", FontType.CALIBRI, 16, ColorType.Blue());
        document.AddText(" Next text!", FontType.CALIBRI, 12, ColorType.Red());

        document.Save();
        document.Show();

        Document document_2 = new Document("HelloWorldCS5800_2");
        document_2.Load("HelloWorldCS5800");
        document_2.AddText(" What 4th?", FontType.VERDANA, 16, ColorType.Blue());
        document_2.Save();

        document_2.Show();
    }
}